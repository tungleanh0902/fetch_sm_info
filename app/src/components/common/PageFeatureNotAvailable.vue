<template>
  <TmPage data-title="Feature Not Available">
    <FeatureNotAvailable :feature="feature" />
  </TmPage>
</template>

<script>
import TmPage from "common/TmPage"
import FeatureNotAvailable from "common/FeatureNotAvailable"

export default {
  name: `page-not-available`,
  components: {
    TmPage,
    FeatureNotAvailable,
  },
  props: {
    feature: {
      type: String,
      required: true,
    },
  },
}
</script>
