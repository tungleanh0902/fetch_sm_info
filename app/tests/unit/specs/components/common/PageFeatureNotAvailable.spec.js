import { shallowMount } from "@vue/test-utils"
import PageFeatureNotAvailable from "common/PageFeatureNotAvailable"

describe(`PageFeatureNotAvailable`, () => {
  let wrapper, $store

  beforeEach(() => {
    const state = {
      session: {
        address: `cosmos`,
        atoms: 1,
      },
    }
    $store = {
      state,
      getters: {
        connected: true,
      },
    }
    wrapper = shallowMount(PageFeatureNotAvailable, {
      mocks: {
        $store,
      },
      propsData: {
        feature: "spacetravel",
      },
    })
  })

  it(`shows a warning about a feature not being available`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
