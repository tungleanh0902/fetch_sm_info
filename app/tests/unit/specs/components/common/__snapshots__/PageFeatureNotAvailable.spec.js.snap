// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFeatureNotAvailable shows a warning about a feature not being available 1`] = `
<tmpage-stub
  data-title="Feature Not Available"
  emptysubtitle=""
  emptytitle="No data"
  loaderpath=""
>
  <featurenotavailable-stub
    feature="spacetravel"
  />
</tmpage-stub>
`;
